### PR TITLE
Use flags for URLs

### PR DESCRIPTION
**Pull request by tempoz**
Introduces URLFlags to allow URL parameters to be parsed on startup, which means we don't have
error check them everytime we parse them in the future.

These URLs are used in many places, so we introduce packages for the flags and assocaited code
to reside in, to give everything that uses them (and only things that use them) common
definitions of the flags.

---

**Version bump**: Minor

### CODE REVIEW

**Comment by tempoz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1771** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1771" target="_blank">Graphite</a> 
    * **PR #1772** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1772" target="_blank">Graphite</a> 
      * **PR #1773** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1773" target="_blank">Graphite</a> 
        * **PR #1774** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1774" target="_blank">Graphite</a>  (this PR)
          * **PR #1775** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1775" target="_blank">Graphite</a> 
            * **PR #1776** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1776" target="_blank">Graphite</a> 
              * **PR #1777** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1777" target="_blank">Graphite</a> 
                * **PR #1778** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1778" target="_blank">Graphite</a> 
                  * **PR #1779** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1779" target="_blank">Graphite</a> 
                    * **PR #1780** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1780" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: enterprise/server/auth/auth.go
Comment (vadimberezniker) [inline enterprise/server/auth/auth.go:466]: golang discourages repeating the package name as part of package functions or structs since the caller has to repeat them and the repeated name does not convey anything new. Perhaps a name like `WithPath` would be better?

File: enterprise/server/hostedrunner/hostedrunner.go
Comment (vadimberezniker) [inline enterprise/server/hostedrunner/hostedrunner.go:135]: Same here, can this be simply `String`?

File: server/endpoint_urls/build_buddy_url/build_buddy_url.go
Comment (vadimberezniker) [inline server/endpoint_urls/build_buddy_url/build_buddy_url.go:11]: Consider adding a function that will define the flag and return the string like the standard flag functions.

func URLVar() *string {
 ...
}

**Review by vadimberezniker — APPROVED** [review]

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1774?utm_source=stack-comment).


Job Reference: Qg74nVRI1LSaCBRkxMdN

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1774?utm_source=stack-comment).


Job Reference: IzK4IJkw70q5gig9wHik

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1774?utm_source=stack-comment).


Job Reference: 6MoHHD1cAEi13ZYwKoui

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1774?utm_source=stack-comment).


Job Reference: YXJ0IDeHv03gACfZZh33

**Pull request merged**
